services/pub/pub-points.service.js

```diff
@@ -0,0 +1,58 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+
+const documentation = `<p>A measure of quality. This includes several dimensions of quality such as code style, platform support, and maintainability.</p>`
+
+const keywords = ['dart', 'flutter']
+
+const schema = Joi.object({
+  grantedPoints: Joi.number().min(0).max(130).required(),
+  maxPoints: Joi.number().min(130).max(130).required(),
+}).required()
+
+const title = 'Pub Points'
+
+export default class PubPoints extends BaseJsonService {
+  static category = 'rating'
+
+  static route = { base: 'pub/points', pattern: ':packageName' }
+
+  static examples = [
+    {
+      title,
+      keywords,
+      documentation,
+      namedParams: { packageName: 'analysis_options' },
+      staticPreview: {
+        label: 'points',
+        message: '130/130',
+        color: 'brightgreen',
+      },
+    },
+  ]
+
+  static defaultBadgeData = { label: 'points' }
+
+  static render({ grantedPoints, maxPoints, packageName }) {
+    return {
+      label: 'points',
+      message: `${grantedPoints}/${maxPoints}`,
+      color: 'brightgreen',
+      link: `https://pub.dev/packages/${packageName}`,
+    }
+  }
+
+  async fetch({ packageName }) {
+    return this._requestJson({
+      schema,
+      url: `https://pub.dev/api/packages/${packageName}/score`,
+    })
+  }
+
+  async handle({ packageName }) {
+    const score = await this.fetch({ packageName })
+    const grantedPoints = score.grantedPoints
+    const maxPoints = score.maxPoints
+    return this.constructor.render({ grantedPoints, maxPoints, packageName })
```

Comment:
Is `maxPoints` always 130? We seem to be doing a bit of a random mix of using `maxPoints` in some places and hard-coding `130` in others..

Reply:
Currently, the maximum number of points for the packages or plugins is 130.

Reply:
I guess what I am getting at here is:

* If we expect `maxPoints` can change over time, lets use `maxPoints` as a variable but loosen the validation. If we validate it as `Joi.number().min(130).max(130).required()` then if `maxPoints` ever changes upstream, the badges all break because the validation fails.
* If we're saying it can only ever be 130 then its not really buying us anything for it to be a variable.

Reply:
Okay, so I have made max points dynamic and added nonNegativeInteger validation. So they will not break even if max points change over time.